### ISSUE TITLE
uBlox M8 SFRBX Galileo Navigation

### ISSUE THREAD

**Issue by mbavaro** [tomojitakasu/RTKLIB#157]
At the present moment it is possible to obtain Galileo NAV using RTKCONV.EXE 2.4.3b8.
CONVBIN.EXE does not work as file names are not properly initialized, e.g. 
`char work[1024],ofile_[7][1024],*ofile[7],*p;`
should be instead 
`char work[1024],ofile_[7][1024]={"\0"},*ofile[7]={NULL},*p;`

However, when trying to compile the current sources in 2.4.3b8 with a recent MINGW gcc, convbin.exe is unable to decode Galileo nav messages in SFRBX (note this is present in M8N as well).
The consistency of I/NAV systematically fails in rcvraw.c. I am analyzing the issue (compiler dependent or simply a bug?), but I though it would be worth asking first if it is fixed already elsewhere.

Btw, having changed these lines:
```
        raw->obs.data[n].code[0]=
            sys==SYS_CMP?CODE_L1I:(sys==SYS_GAL?CODE_L1X:CODE_L1C);
```
prevents 2.4.3b8 to generate Galileo OBS (unless set_obstype() in convrnx.c is also changed).

Tomoji, you seem to be ahead of the repository for a few things related to Galileo (mentioning 2.4.3b9 somewhere). Any plans to update Galileo support soon?

**Comment by mbavaro** [comment]
OK, I found it. 

The offending line is 694 of ublox.c:
`    unsigned char *p=raw->buff+6+off,buff[32],crc_buff[25]={0};
`
as really the code goes and sets bits beyond 200 at line 719:

`    for (i=0,j=118;i<11;i++,j+=8) setbitu(crc_buff,j,8,getbitu(buff+16,i*8,8));
`

Maybe the Borland compiler has a different pragma pack, but GCC writes on the first bits of `buff` at the last cycle writing onto `crc_buff`.
So just declaring crc_buff uint8_t[**26**] fixes the problem.

By the way, nasty uBlox describing the Galileo payload as a simple uint32_t[8] when really there is a ninth doubleword (no clue what that keeps), and all words need changing endianness - against all odds.

Will mark as closed as soon as the fix is applied to the repo.

**Comment by tomojitakasu** [comment]
Thanks. It is fixed in 2.4.3 b10.
